include/groonga/pat.h

```diff
@@ -173,6 +173,28 @@ grn_pat_cursor_delete(grn_ctx *ctx,
     }                                                                          \
   } while (0)
 
+#define GRN_PAT_EACH_BEGIN(ctx, pat, cursur, id, key, key_size, value)         \
+  do {                                                                         \
+    grn_pat_cursor *cursur = NULL;                                             \
+    if (pat && grn_pat_size(ctx, pat) > 0) {                                   \
+      cursur = grn_pat_cursor_open(ctx, pat, NULL, 0, NULL, 0, 0, -1, 0);      \
+    }                                                                          \
+    if (cursur) {                                                              \
```

Comment:
`GRN_TABLE_EACH_BEGIN()`みたいにこうしない？

```suggestion
    if (pat && grn_pat_size(ctx, pat) > 0) {                                   \
      grn_pat_cursor *cursur = grn_pat_cursor_open(ctx, pat, NULL, 0, NULL, 0, 0, -1, 0);      \
      if (cursur) {                                                              \
```